clients/src/main/java/org/apache/kafka/clients/admin/internals/RemoveMembersFromConsumerGroupHandler.java

```diff
@@ -0,0 +1,133 @@
+
+    private final CoordinatorKey groupId;
+    private final List<MemberIdentity> members;
+    private final Logger log;
+    private final AdminApiLookupStrategy<CoordinatorKey> lookupStrategy;
+
+    public RemoveMembersFromConsumerGroupHandler(
+        String groupId,
+        List<MemberIdentity> members,
+        LogContext logContext
+    ) {
+        this.groupId = CoordinatorKey.byGroupId(groupId);
+        this.members = members;
+        this.log = logContext.logger(RemoveMembersFromConsumerGroupHandler.class);
+        this.lookupStrategy = new CoordinatorStrategy(CoordinatorType.GROUP, logContext);
+    }
+
+    @Override
+    public String apiName() {
+        return "leaveGroup";
+    }
+
+    @Override
+    public AdminApiLookupStrategy<CoordinatorKey> lookupStrategy() {
+        return lookupStrategy;
+    }
+
+    public static AdminApiFuture.SimpleAdminApiFuture<CoordinatorKey, Map<MemberIdentity, Errors>> newFuture(
+        String groupId
+    ) {
+        return AdminApiFuture.forKeys(Collections.singleton(CoordinatorKey.byGroupId(groupId)));
+    }
+
+    @Override
+    public LeaveGroupRequest.Builder buildRequest(int coordinatorId, Set<CoordinatorKey> keys) {
+        return new LeaveGroupRequest.Builder(groupId.idValue, members);
+    }
+
+    @Override
+    public ApiResult<CoordinatorKey, Map<MemberIdentity, Errors>> handleResponse(
+        Node coordinator,
+        Set<CoordinatorKey> groupIds,
+        AbstractResponse abstractResponse
+    ) {
+        final LeaveGroupResponse response = (LeaveGroupResponse) abstractResponse;
+        Map<CoordinatorKey, Map<MemberIdentity, Errors>> completed = new HashMap<>();
+        Map<CoordinatorKey, Throwable> failed = new HashMap<>();
+        List<CoordinatorKey> unmapped = new ArrayList<>();
+
+        final Errors error = Errors.forCode(response.data().errorCode());
+        if (error != Errors.NONE) {
+            handleError(groupId, error, failed, unmapped);
+        } else {
+            final Map<MemberIdentity, Errors> memberErrors = new HashMap<>();
+            for (MemberResponse memberResponse : response.memberResponses()) {
+                memberErrors.put(new MemberIdentity()
+                                     .setMemberId(memberResponse.memberId())
+                                     .setGroupInstanceId(memberResponse.groupInstanceId()),
+                                 Errors.forCode(memberResponse.errorCode()));
+
+            }
+            completed.put(groupId, memberErrors);
+        }
+        return new ApiResult<>(completed, failed, unmapped);
+    }
+
+    private void handleError(
+        CoordinatorKey groupId,
+        Errors error, Map<CoordinatorKey,
+        Throwable> failed,
+        List<CoordinatorKey> unmapped
+    ) {
+        switch (error) {
+            case GROUP_AUTHORIZATION_FAILED:
+                log.error("Received authorization failure for group {} in `DeleteConsumerGroupOffsets` response", groupId,
```

Comment:
Another place where I think we want this to be `RemoveMembersFromConsumerGroup` instead of `DeleteConsumerGroupOffsets`